### PR TITLE
Authentication using different API keys via the client without race conditions

### PR DESCRIPTION
**Pull request by Pbarbosa92**
This PR allows the api key to be changed at runtime.

inductiva.api_key no longer exists and was replaced by inductiva._api_key that should not be accessed by the user.
Instead the user should use inductiva.set_api_key("key") and inductiva.get_api_key()

We can now do 
```
#Uses the api key from env vars

import inductiva

# Set simulation input directory
input_dir = inductiva.utils.download_from_url(
    "https://storage.googleapis.com/inductiva-api-demo-files/"
    "openfast-input-example.zip", unzip=True)

# List of commands to run
commands = [
    "openfast IEA-15-240-RWT-Monopile.fst"
]

# Initialize OpenFAST simulator
openfast = inductiva.simulators.OpenFAST()

# Run simulation 
task = openfast.run(input_dir=input_dir,
                    commands=commands)

task.wait()
task.download_outputs()

inductiva.set_api_key("api_key_2")

# Set simulation input directory
input_dir = inductiva.utils.download_from_url(
    "https://storage.googleapis.com/inductiva-api-demo-files/"
    "openfast-input-example.zip", unzip=True)

# List of commands to run
commands = [
    "openfast IEA-15-240-RWT-Monopile.fst"
]

# Initialize OpenFAST simulator
openfast = inductiva.simulators.OpenFAST()

# Run simulation 
task = openfast.run(input_dir=input_dir,
                    commands=commands)

task.wait()
task.download_outputs()
```

This new method should work in a multithreaded environment.

### CODE REVIEW

**Comment by Pbarbosa92** [comment]
**QA**

In order to test this feature there are a few steps to follow:

1. Create 4 new accounts (use mail aliases)
2. Get 4 api keys (1 for each new account)
3. Paste the api keys in the snippet bellow
4. Run the snippet
5. Check if 100 tasks were created for each api key
```

import inductiva
from threading import Thread

input_dir = inductiva.utils.download_from_url(
        "https://storage.googleapis.com/inductiva-api-demo-files/"
        "fds-input-example.zip", unzip=True)

def test_two_keys(key1,key2,input_dir):
    for _ in range(100):
        inductiva.set_api_key(key1)

        fds = inductiva.simulators.FDS()
        task = fds.run(input_dir=input_dir,
                    sim_config_filename="mccaffrey.fds",
                    post_processing_filename="mccaffrey.ssf",
                    n_vcpus=1)
        inductiva.set_api_key(key2)

        fds = inductiva.simulators.FDS()
        task = fds.run(input_dir=input_dir,
                    sim_config_filename="mccaffrey.fds",
                    post_processing_filename="mccaffrey.ssf",
                    n_vcpus=1)

api_key_2 = "api_key_2"
api_key_3 = "api_key_3"
api_key_4 = "api_key_4"
api_key_5 = "api_key_5"

threads = []

t = Thread(target=test_two_keys, args=(api_key_2, api_key_3,input_dir))
t.start()
threads.append(t)

t = Thread(target=test_two_keys, args=(api_key_4, api_key_5,input_dir))
t.start()
threads.append(t)

for t in threads:
    t.join()

```

File: inductiva/resources/machines_base.py [hunk 1]
```diff
@@ -71,10 +71,6 @@ def __init__(self,
         #the request machine_groups.get()
         self._active_machines = 0
         self.num_machines = 0
-
-        # Set the API configuration that carries the information from the client
-        # to the backend.
-        self._api = compute_api.ComputeApi(api.get_client())
```
[the remark this answers is not in the document]
Comment (Pbarbosa92) [reply inductiva/resources/machines_base.py]: Hmmm. I see your point. I'm reverting this changes.

File: inductiva/__init__.py [hunk 2]
```diff
@@ -70,17 +73,23 @@ def _check_for_available_package_update():
...
 
 def _check_key():
-    global _checked_key
-
-    if not _checked_key and utils.format_utils.getenv_bool(
-            "GITHUB_ACTIONS", False) is not True:
-        api.methods.validate_api_key(api_key)
-        _checked_key = True
+    if utils.format_utils.getenv_bool("GITHUB_ACTIONS", False) is not True:
+        api.methods.validate_api_key()
```
Comment (sergiomsantos) [inline inductiva/__init__.py:77]: Why remove the verification?

Also, and in this context, `X is not True` is the same as `not X`

File: inductiva/api/methods.py [hunk 3]
```diff
@@ -29,9 +29,12 @@
...
 
-def validate_api_key(api_key: Optional[str]) -> Configuration:
+def validate_api_key() -> Configuration:
     """Validates the API key and returns API configuration"""
-    if inductiva.api_key is None:
+
+    api_key = inductiva.get_api_key()
+
+    if api_key is None:
         # pylint: disable=line-too-long
```
Comment (sergiomsantos) [inline inductiva/api/methods.py:38]: do we need these directives for pylint in this section?

File: inductiva/api/methods.py [hunk 4]
```diff
@@ -29,9 +29,12 @@
 from inductiva.utils import format_utils, files
 
 
-def validate_api_key(api_key: Optional[str]) -> Configuration:
+def validate_api_key() -> Configuration:
```
Comment (sergiomsantos) [inline inductiva/api/methods.py:32]: let's rename this method. It is not validating the key - it is building an  api_config object.
Validation is only a subproduct of the process.

File: inductiva/tasks/streams.py
Comment (sergiomsantos) [inline inductiva/tasks/streams.py:1]: Why changes in this file?

File: inductiva/__init__.py [hunk 5]
```diff
@@ -70,17 +73,23 @@ def _check_for_available_package_update():
...
 
 
 _check_for_available_package_update()
 
 
+def set_api_key(api_key):
+    """Sets the value of `inductiva._api_key` to `api_key"""
+    _api_key.set(api_key)
```
Comment (sergiomsantos) [inline inductiva/__init__.py:85]: maybe we can perform the validation here?

**Review by sergiomsantos — APPROVED** [review]

**Pull request merged**